Added Image/Gif Viewer

Foso opened the pull request:
Hi, i've created an image/gif viewer for amaze.

Functions:
Used [TouchImageView](https://github.com/MikeOrtiz/TouchImageView) for 
* Pinch to zoom
* Double Tap to zoom

Used [Glide](https://github.com/bumptech/glide) for GIF support

Added:
* Set image as Wallpaper (or whatever)
* Delete Image
* Share Image
* Rotate Image Left
* intent filter to handle intent from other apps
* show image properties (not sure, which are really relevant?)
* single tap to hide/show toolbar
* swipe left/right to show next/previous image

The bottom toolbar needs some design improvements.

I'm open for any suggestions.
![amaze image viewer](https://user-images.githubusercontent.com/5015532/29141356-3a4865c2-7d4e-11e7-83db-0529ba0f33ed.png)

EmmanuelMess commented:
For ic_share_white_24dp.png please use a vector.

EmmanuelMess commented on app/src/main/res/values/strings.xml:
```diff
@@ -682,4 +682,6 @@
     <string name="failed_cloud_new_connection">Failed to add new Cloud connection</string>
     <string name="failed_cloud_api_key">Failed to authenticate Cloud plug-in</string>
     <string name="local_inet_addr_error">Error getting local inet address</string>
+    <string name="setImageAs">Set image as:</string>
```
Use underscores for string resource names.

EmmanuelMess commented on app/src/main/res/menu/image_viewer_menu.xml:
```diff
@@ -0,0 +1,49 @@
+
+    <item
+        android:id="@+id/rotate_left"
+        android:title="RotateLeft"
```
You forgot to set this as a resource

EmmanuelMess commented:
With this image:
![687474703a2f2f692e696d6775722e636f6d2f765963636e32412e706e67](https://user-images.githubusercontent.com/10991116/29146088-f777c246-7d35-11e7-82be-52a0c0bd60e8.png)
I'm seeing a height and width of 0.

EmmanuelMess commented on app/src/main/java/com/amaze/filemanager/activities/ImageViewerActivity.java:
```diff
@@ -0,0 +1,315 @@
+  }
+
+  private void updateTvFolderName() {
+    tvFolderName.setText("Folder:"
```
Hardcoded "Folder:", are you sure? Also check [this](https://stackoverflow.com/questions/3656371/dynamic-string-using-string-xml) out.

vishal0071 commented:
Does this also include gif playing in the icon itself?

Foso commented:
Hi @EmmanuelMess  
thank you for your review, sorry for the issues.
I extracted the strings. 
I changed the share icon to vector.
I made a few optimizations. 
Your example image shows a height and width of 0, because i'm reading the exif information and this image doesn't have any.

Hi @vishal0071
with Glide you can play gif in any ImageView

EmmanuelMess commented:
@Foso if you don't have information, don't assume that the image is 0x0, you can check for the information "manualy" or not show the fields in properties.

EmmanuelMess commented on app/src/main/java/com/amaze/filemanager/activities/ImageViewerActivity.java:
```diff
@@ -0,0 +1,298 @@
+    private void loadImage() {
+        GlideApp.with(ivImage)
+                .load(currentFile.getPath())
+                .error(R.mipmap.ic_launcher)
```
Does this mean that ic_launcher will be shown if Glide fails to load the correct one? You should snackbar an error like "Error loading image" and show a grey/white cross.

vishal0071 commented:
Pretty solid feature, can't merge on master as it is quite incomplete as of now. But we can create a new branch and help you with this @Foso What do you think?

EmmanuelMess commented:
You also should rebase onto the last commit in master.

Alexander-- commented on app/src/main/java/com/amaze/filemanager/filesystem/FileUtil.java:
```diff
@@ -1072,5 +1072,29 @@ public OutputStream write(long size)
+        Cursor cursor;
+        if (uri != null) {
+            if ("content".equals(uri.getScheme())) {
+                cursor = c.getContentResolver().query(uri, new String[]{MediaStore.Images.ImageColumns.DATA}, null, null, null);
```
Have you tested this code with Google Drive and other third-party ContentProviders? AFAIK, using value of `_data` column like this does not really work with any `content://` Uris except ones from media storage (and media storage only works, because it's files are stored in public location on sdcard).

EmmanuelMess commented:
I am closing this until you decide to update the code.
